apis/apps/v1alpha1/type.go

```diff
@@ -1288,3 +1291,66 @@ const (
 	LogsKind    = "logs"
 	TracesKind  = "traces"
 )
+
+type ServiceMonitorTemplate struct {
+	// Standard k8s object's metadata.
+	//
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// ServiceMonitorSpec contains specification parameters for a ServiceMonitor
+	//
+	// +optional
+	ServiceMonitorSpec monitoringv1.ServiceMonitorSpec `json:"serviceMonitorSpec,omitempty"`
+}
+
+type VMMonitorTemplate struct {
+	// Standard k8s object's metadata.
+	//
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// VMServiceScrapeSpec defines the desired state of VMServiceScrape
+	//
+	// +optional
+	VMServiceScrapeSpec vmv1beta1.VMServiceScrapeSpec `json:"vmServiceScrapeSpec,omitempty"`
+}
+
+type MonitorIntegration struct {
+	// Specifies the ServiceMonitor template used in the Component.
+	//
+	// Example usage:
+	// ```yaml
+	// name: prometheus-scrape
+	// namespace: default
+	// labels:
+	//   k8s-app: node-exporter
+	// serviceMonitorTemplate:
+	//   selector:
+	//     matchLabels:
+	//       app: app-exporter
+	//       k8s-app: app-exporter
+	//   endpoints:
+	//   - port: metrics_port
+	//   jobLabel: k8s-app
+	// ```
+	//
+	// +optional
+	ServiceMonitorTemplate *ServiceMonitorTemplate `json:"serviceMonitorTemplate,omitempty"`
+
+	// Specifies the VMServiceScrape template used in the Component.
```

Comment:
leave a comment to tell user that integration to vm is currently not supported yet.

Reply:
//TODO: will be ignored by api-doc